src/Pages/Capture.elm

```diff
@@ -181,18 +181,33 @@ update msg model =
                             ( { model | error = "Error while starting the timer" }, Cmd.none )
 
         ToggleCompleted capture _ ->
-            let
-                captureStatus =
-                    if capture.status == Completed then
-                        ToDo
+            case capture.status of
+                InProgress ->
+                    case getCurrentTimer capture.timers of
+                        Just t ->
+                            ( model
+                            , Cmd.batch
+                                [ stopTimer (token model.session) t.idTimer capture.idCapture
+                                , apiUpdateCapture
+                                    (token model.session)
+                                    { capture | status = Completed }
+                                ]
```

Comment:
If the capture is in progress, two requests are sent, the first one to stop the timer then the second one to update the status of the capture to "completed"

Reply:
@SimonLab this should probably be a single request. 💭 
Stopping a timer should update the status to `completed`.  ✅

Reply:
We can add multiple timers to the capture, i.e we can start/stop timers multiple time.
Only when the checkbox is checked the capture is then defined as `completed`.

However I can update the `apiUpdateCapture` to verify if the most recent timer is still running and stop it if the capture status is `completed`

Reply:
@SimonLab right but the request to update the status to `completed` should _also_ `stop` the `timer` so we should only need to make a _single_ HTTP request which will speed up UX.
Not important right now, this can be optimised later. 👍